gui/src/app/state/mod.rs

```diff
@@ -82,34 +86,50 @@ impl State for Home {
             &Menu::Home,
             cache,
             None,
-            view::home::home_view(&self.balance, &self.pending_events, &self.events),
+            view::home::home_view(
+                &self.balance,
+                self.recovery_warning.as_ref(),
+                self.recovery_alert.as_ref(),
+                &self.pending_events,
+                &self.events,
+            ),
         )
     }
 
     fn update(
         &mut self,
         daemon: Arc<dyn Daemon + Sync + Send>,
-        _cache: &Cache,
+        cache: &Cache,
         message: Message,
     ) -> Command<Message> {
         match message {
             Message::Coins(res) => match res {
                 Err(e) => self.warning = Some(e),
                 Ok(coins) => {
                     self.warning = None;
-                    self.balance = Amount::from_sat(
-                        coins
-                            .iter()
-                            .map(|coin| {
-                                // If the coin is not spent and is its transaction is confirmed
-                                if coin.spend_info.is_none() && coin.block_height.is_some() {
-                                    coin.amount.to_sat()
-                                } else {
-                                    0
-                                }
-                            })
-                            .sum(),
-                    );
+                    self.balance = Amount::from_sat(0);
+                    let mut recovery_warning = (Amount::from_sat(0), 0);
+                    let mut recovery_alert = (Amount::from_sat(0), 0);
+                    for coin in coins {
+                        if coin.spend_info.is_none() && coin.block_height.is_some() {
+                            self.balance += coin.amount;
+                            let timelock = daemon.config().main_descriptor.timelock_value();
+                            let seq = remaining_sequence(&coin, cache.blockheight as u32, timelock);
+                            if seq == 0 {
+                                recovery_alert.0 += coin.amount;
+                                recovery_alert.1 += 1;
+                            } else if seq < timelock * 10 / 100 {
```

Comment:
Why `* 10 / 100` instead of just `/ 10`?

Also technically it'd be more correct to
```suggestion
                            } else if seq * 10 < timelock {
```
So that it doesn't truncate. But no big deal.